### DISCUSSION TITLE
Should reusable enums display the enum type or the underlying data type?

### DISCUSSION THREAD

**Discussion by jzajac** [scalar/scalar#3811] [Q&A] [answered]
I'm not sure if this is a bug report, feature request, or user error. 

OpenAPI 3.0 supports [reusable enums](https://swagger.io/docs/specification/v3_0/data-models/enums/#reusable-enums) which appear under the `components` section.

In Scalar, when viewing an OpenAPI spec that contains reusable enums, the UI only indicates the field is an enum with string values.  I would expect it to indicate the field is a reusable enum, perhaps even linking to the model. 

[Here's a simplified example](https://sandbox.scalar.com/e/iPsIC) OpenAPI spec which demonstrates this behavior. Scalar successfully imported `MyReusableEnum` and is correctly pulling in the values from that enum, but the context of where those values came from gets lost. See screenshot below. 

![Screenshot 2024-11-08 at 6 24 21 AM](https://github.com/user-attachments/assets/8e18aa39-34ac-45a7-b2d9-7fc03b4bdd84)

When I download the OpenAPI spec from Scalar, the doc correctly points back to the reusable enum. So I don't think this is a case where metadata is getting lost when importing OpenAPI specs. 

For comparison, importing the same OpenAPI spec above into the [Swagger Editor](https://editor.swagger.io/), this does indicate the field is a reusable enum. See screenshot below. 

![Screenshot 2024-11-08 at 6 30 26 AM](https://github.com/user-attachments/assets/8ee73ae7-beff-48b9-ac7a-3ba575103a07)

**Comment by hanspagel** [comment, marked as answer]
Yes! Same as #1847

**Reply by jzajac** [reply]
Thanks @hanspagel! Do you have a sense of where that issue falls on the Scalar roadmap?
